client/java/transports-dataplex/src/main/java/io/openlineage/client/transports/dataplex/DataplexTransport.java

```diff
@@ -0,0 +1,132 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.client.transports.dataplex;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.rpc.FixedHeaderProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.datacatalog.lineage.v1.LineageSettings;
+import com.google.cloud.datacatalog.lineage.v1.ProcessOpenLineageRunEventRequest;
+import com.google.cloud.datalineage.producerclient.helpers.OpenLineageHelper;
+import com.google.cloud.datalineage.producerclient.v1.SyncLineageClient;
+import com.google.cloud.datalineage.producerclient.v1.SyncLineageProducerClient;
+import com.google.cloud.datalineage.producerclient.v1.SyncLineageProducerClientSettings;
+import com.google.protobuf.Struct;
+import io.openlineage.client.OpenLineage;
+import io.openlineage.client.OpenLineageClientException;
+import io.openlineage.client.OpenLineageClientUtils;
+import io.openlineage.client.transports.Transport;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DataplexTransport extends Transport implements Closeable {
+
+  private final SyncLineageClient producerClient;
+
+  private final String parent;
+
+  public DataplexTransport(@NonNull DataplexConfig config) throws IOException {
+    this(config, SyncLineageProducerClient.create(createSettings(config)));
+  }
+
+  protected DataplexTransport(@NonNull DataplexConfig config, @NonNull SyncLineageClient client)
+      throws IOException {
+    this.producerClient = client;
+    this.parent =
+        String.format(
+            "projects/%s/locations/%s",
+            getProjectId(config, createSettings(config)),
+            config.getLocations() != null ? config.getLocations() : "us");
+  }
+
+  @Override
+  public void emit(OpenLineage.@NonNull RunEvent runEvent) {
+    emitEvent(runEvent);
+  }
+
+  @Override
+  public void emit(OpenLineage.@NonNull DatasetEvent datasetEvent) {
+    emitEvent(datasetEvent);
+  }
+
+  @Override
+  public void emit(OpenLineage.@NonNull JobEvent jobEvent) {
+    emitEvent(jobEvent);
+  }
+
+  private <T extends OpenLineage.BaseEvent> void emitEvent(T event) {
+    try {
+      String eventJson = OpenLineageClientUtils.toJson(event);
+      Struct openLineageStruct = OpenLineageHelper.jsonToStruct(eventJson);
+      ProcessOpenLineageRunEventRequest request =
+          ProcessOpenLineageRunEventRequest.newBuilder()
+              .setParent(parent)
+              .setOpenLineage(openLineageStruct)
+              .build();
+      producerClient.processOpenLineageRunEvent(request);
```

Comment:
@midamkina Should we also inspect the contents of the response to establish whether the request was successful or not, or is it ok to just rely on an exception to be raised in case of failure?